osu.Game.Rulesets.Mania/Mods/ManiaModPlayfieldCover.cs

```diff
@@ -27,22 +35,50 @@ public abstract class ManiaModPlayfieldCover : ModHidden, IApplicableToDrawableR
         [SettingSource("Coverage", "The proportion of playfield height that notes will be hidden for.")]
         public abstract BindableNumber<float> Coverage { get; }
 
+        [SettingSource("Change coverage based on combo", "Increase the coverage as combo increases.")]
+        public BindableBool ComboBasedCoverage => new BindableBool(true);
+
+        protected readonly BindableInt Combo = new BindableInt();
+
+        public override void ApplyToScoreProcessor(ScoreProcessor scoreProcessor)
+        {
+            Combo.BindTo(scoreProcessor.Combo);
+
+            // Default value of ScoreProcessor's Rank in Hidden/Fadein Mod should be SS+
+            scoreProcessor.Rank.Value = ScoreRank.XH;
```

Comment:
The way to do this properly is by calling base:

```suggestion
            base.ApplyToScoreProcessor(scoreProcessor);
```